src/deserialize_sequence.cc

```diff
@@ -224,7 +226,34 @@ void extract_disabled(gul14::string_view extract, Step& step)
     step.set_disabled(val);
 }
 
-} // namespace anonymous
+} // anonymous namespace
+
+SequenceInfo get_sequence_info_from_filename(gul14::string_view filename)
+{
+    SequenceInfo result;
+
+    const std::string str = unescape_filename_characters(filename);
+
+    auto opening_bracket = str.rfind('[');
+    auto closing_bracket = str.find(']', opening_bracket);
+    if (opening_bracket != filename.npos && closing_bracket == str.size() - 1)
+    {
+        result.name = SequenceName::from_string(
+            gul14::trim(str.substr(0, opening_bracket)));
+        result.unique_id = UniqueId::from_string(
+            str.substr(opening_bracket + 1, closing_bracket - opening_bracket - 1));
+
+        if (result.name && result.unique_id)
+            return result;
+    }
+
+    // The filename is not in the new format "name[uid]". We assume it is just the label.
```

Comment:
See above, this should fail - at least after the grace/transition period. Then this would be a malformed name.  
Please let not add legacy cruft already now.